registry/storage/tagstore.go

```diff
@@ -132,46 +148,117 @@ func (ts *tagStore) linkedBlobStore(ctx context.Context, tag string) *linkedBlob
 	}
 }
 
+type atomicError struct {
```

Comment:
can we move the implementation of this type at the bottom of this file so it doesn't "interrupt" the [reading] flow of `TagStore` implementation.